### ISSUE TITLE
Add a warning if contnuum-normalized analysis functions get non-dimensionless spectra

### ISSUE THREAD

**Issue by eteq** [astropy/specutils#547]
In #538 a continuum-subtraction checker was added to catch the case where a user runs an analysis function that expects a zero-baseline spectrum on a spectrum where continuum has not been subtracted.

In #535 we discussed doing something similar for functions requiring continuum normalization (right now I think that's just `equivalent_width`), but it occurs to me there's an easier heuristic for continuum-normalization: the unit has to be dimensionless because the continuum and spectrum should have the same units so dividing them out gives dimensionless.  So perhaps we should just check the unit and raise a warning if it's not dimensionless?  That has essentially zero performance penalty (unlike the continuum-checker...)

cc @camipacifici (since it was your use case that originally led to this idea)

**Comment by eteq** [comment]
Update from #546 - results in https://github.com/astropy/specutils/issues/546#issuecomment-556523323 .

TL;DR is that it's a small (2x or less) effect for functions that operate on the whole spectrum, but potentially very large (~10x) for very large spectra/cubes where you want to do a small operation on only some of them.

So that's an issue.  We may have to pick and choose which the decorators get. Probably needs more discussion before implementing beyond #538

**Comment by eteq** [comment]
I've collated the list of potential places for this decorator.  Below when I say "local" I mean that the measurement is typically in a small spectral region compared to the whole spectrum, whereas "global" means the function operates on the whole spectrum.  Relevant because the "local" ones are the ones where the slow-down could be order-of-magnitude and gets worse with spectrum size (because the continuum-checker is a "global" algorithm), whereas the global ones generally get better with spectrum size (based on #546, of order ~2x slowdown)

- [ ] [centroid](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.centroid.html#specutils.analysis.centroid) - continuum subtraction required, "local" measurement
- [ ] [equivalent_width](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.equivalent_width.html#specutils.analysis.equivalent_width)  - continuum normalization required, "local" measurement
- [ ] [fwhm](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.fwhm.html#specutils.analysis.fwhm)  - continuum subtraction required, "local" measurement
- [ ] [fwzi](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.fwzi.html#specutils.analysis.fwzi)  - continuum subtraction required, "local" measurement
- [ ] [gaussian_fwhm](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.gaussian_fwhm.html#specutils.analysis.gaussian_fwhm)  - continuum subtraction required, "local" measurement
- [ ] [gaussian_sigma_width](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.gaussian_sigma_width.html#specutils.analysis.gaussian_sigma_width)  - continuum subtraction required, "local" measurement
- [ ] [line_flux](https://specutils.readthedocs.io/en/stable/api/specutils.analysis.line_flux.html#specutils.analysis.line_flux)  - continuum subtraction *usually* intended, "local" measurement
- [ ] [find_lines_threshold](https://specutils.readthedocs.io/en/stable/api/specutils.fitting.find_lines_threshold.html#specutils.fitting.find_lines_threshold)  - continuum subtraction required, "global" measurement
- [ ] [fit_lines](https://specutils.readthedocs.io/en/stable/api/specutils.fitting.fit_lines.html#specutils.fitting.fit_lines)  - continuum subtraction sometimes required sometimes not (depends on whether the model has a continuum built-in), "global" *or* "local" measurement

So the question is: which, if any, do we want the continuum checker implemented in as a decorator by default? My first instinct is only `find_lines_threshold`, since that was @camipacifici's original case where there was an issue, and the only definitively global one I found.  But I'm interested in other viewpoints -  @camipacifici (as the inspiration of the original issue), or @keflavich @dhomeier @crawfordsm   (as random potential users), do you have any thoughts here?

**Comment by eteq** [comment]
Oops!  The above comments were in the wrong issue.  Copying them to #548 and striking through the above. - the only relevant one for this issue is `equivalent width)
